ops/main.py

```diff
@@ -30,24 +30,48 @@ def _get_charm_dir():
 def _load_metadata(charm_dir):
     with open(charm_dir / 'metadata.yaml') as f:
         metadata = yaml.load(f, Loader=yaml.SafeLoader)
+
+    functions_meta = charm_dir / 'functions.yaml'
+    functions_dir = charm_dir / 'functions'
+    actions_meta = charm_dir / 'actions.yaml'
+    actions_dir = charm_dir / 'actions'
+    if functions_meta.exists():
+        if 'min-juju-version' not in metadata:
```

Comment:
I think this will end up requiring everybody to know about the `min-juju-version` requirement for functions long past 2.6.x will go EOL.

Maybe we could do a juju version check via `JUJU_VERSION` environment variable and log a warning here instead.

Reply:
We will need something like https://github.com/canonical/operator/pull/100 in order to do version Juju checking and comparison.

I suggest just dropping this check for now.